Conformity/custom-rule/custom-rule.py

```diff
@@ -0,0 +1,269 @@
+			yaml.dump(data=customRule["data"], stream=ruleWriter, indent=4, sort_keys=False)
+
+			print(f'Custom rule saved in file {ruleId}.yaml')
+
+	pass
+
+def get(args):
+	header, region = readConfiguration()
+	endpoint = f'{apiEndpoint.format(region)}/{args.ruleId}' 
+
+	response = requests.get(url=endpoint, headers=header)
+	rule = response.json()
+	ruleId = rule["data"][0]["id"]
+	ruleFilename = f'{ruleId}.yaml'
+
+	with open(f'{workspaceFolder}/{ruleFilename}', 'w') as configWriter:
+		yaml.dump(data=rule["data"][0], stream=configWriter, indent=4, sort_keys=False)
+
+	print(f'Rule saved to file {ruleFilename}')
+	pass
+
+def update(args):
+	header, region = readConfiguration()
+	updatedRule = {}
+
+	with open(f'{workspaceFolder}/{args.file}', 'r') as ruleReader:
+		rule = yaml.safe_load(stream=ruleReader)
+
+		endpoint = f'{apiEndpoint.format(region)}/{rule.get("id")}' 
+		
+		response = requests.put(url=endpoint, json=rule["attributes"], headers=header)
+		updatedRule = response.json()
+	
+	ruleId = updatedRule["data"]["id"]
+	ruleFilename = f'{ruleId}.yaml'
+
+	with open(f'{workspaceFolder}/{ruleFilename}', 'w') as configWriter:
+		yaml.dump(data=updatedRule["data"], stream=configWriter, indent=4, sort_keys=False)
+
+	print(f'Rule saved to file {ruleFilename}')
+
+	pass
+
+def delete(args):
+	header, region = readConfiguration()
+	endpoint = f'{apiEndpoint.format(region)}/{args.ruleId}'
+	
+	response = requests.delete(url=endpoint, headers=header)
+
+	print(f'Rule deleted. Run "custom-rule list" for updated list of rules')
+	pass
+
+def list(args):
+	header, region = readConfiguration()
+	endpoint = apiEndpoint.format(region)
+
+	response = requests.get(url=endpoint, headers=header)
+	rules = response.json()
+
+	with open(f'{workspaceFolder}/rules.yaml', 'w') as configWriter:
+		yaml.dump(data=rules["data"], stream=configWriter, indent=4, sort_keys=False)
+	
+	print("Rules were stored in rule.yaml file.")
+	pass
+
+def run(args):
+	header, region = readConfiguration()
+	endpoint = f'{apiEndpoint.format(region)}/run?accountId={args.accountId}&resourceData=true'
+	
+	rule = {}
+	with open(f'{workspaceFolder}/{args.ruleId}.yaml', 'r') as ruleReader:
+		rule = {
+			"configuration": yaml.safe_load(stream=ruleReader)["attributes"]
+		}
+	
+	rule["configuration"]["resourceId"] = args.resourceId
+	response = requests.post(url=endpoint, json=rule, headers=header)
+
+	dryrunFilename = f'{args.ruleId}.run.yaml'
+	with open(f'{workspaceFolder}/{dryrunFilename}', 'w') as configWriter:
+		yaml.dump(data=response.json(), stream=configWriter, indent=4, sort_keys=False)
+
+	print(f'Run trace saved to file {dryrunFilename}. Note that resourceData information can be used for rule development')
+	pass
+
+def showProviders(args):
+	response = requests.get(url="https://us-west-2.cloudconformity.com/v1/providers")
+	providers = response.json()
+	
+	print(json.dumps([x.get("id") for x in providers["data"]], indent=4))
+	pass
+
+def showServices(args):
```

Comment:
There was a couple of services that didn't show up to me, are you paginating the requests or something like that?

Some examples include: 
[Service Quotas](https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/ServiceQuotas/)
[Managed Streaming for Apache Kafka](https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/MSK/)

Are those included in the *Miscellaneous* ?

Reply:
The issue here is a bit different. We don't yet have rules for MSK and Service Quotas so they are not coming up in the API. 

I think this particular issue has to be closed.